### PR TITLE
refactor: Update ListItem based on latest design

### PR DESCRIPTION
**Pull request by brianacnguyen**
## **Description**
- Moved padding and borderRadius to style
- Updated default verticalAlignment to center

## **Screenshots/Recordings**
Simulator Screen Shot - iPhone 14 Pro - 2023-11-09 at 08 10 47
Simulator Screen Shot - iPhone 14 Pro - 2023-11-09 at 10 19 51
Simulator Screen Shot - iPhone 14 Pro - 2023-11-09 at 10 20 15
Simulator Screen Shot - iPhone 14 Pro - 2023-11-09 at 10 24 56

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by tommasini — COMMENTED** [review]
Hey Brian amazing work, can you record a video of the permissions components? I can see that this impacts the permissions sheets for connecting one or multiple accounts to a dapp

File: app/component-library/components/List/ListItem/ListItem.tsx [hunk 1]
```diff
@@ -21,32 +19,30 @@ import {
...
-              style={{ width: gap }}
-              testID={TESTID_LISTITEM_GAP}
-              accessible={false}
-            />
-          )}
-          {child}
-        </>
-      ))}
+      {React.Children.toArray(children)
+        .filter((child) => !!child)
```
[the remark this answers is not in the document]
Comment (brianacnguyen) [reply app/component-library/components/List/ListItem/ListItem.tsx:33]: its to filter out empty nodes

**Comment by brianacnguyen** [comment]
> Hey Brian amazing work, can you record a video of the permissions components? I can see that this impacts the permissions sheets for connecting one or multiple accounts to a dapp

done

**Review by tommasini — APPROVED** [review]
LGTM! Please merge it after the E2E pass please!

**Pull request merged**